wpilibj/src/main/java/edu/wpi/first/wpilibj/smartdashboard/SendableChooser.java

```diff
@@ -154,6 +168,12 @@ public void initSendable(NTSendableBuilder builder) {
           m_mutex.lock();
           try {
             m_selected = val;
+            if (!m_selected.equals(m_previousVal) && m_listener != null) {
+              m_mutex.unlock();
+              m_listener.accept(m_map.get(val));
+              m_mutex.lock();
+            }
+            m_previousVal = val;
```

Comment:
Unlocking and relocking isn't safe due to the accessing of m_listener and m_map while the lock is released.  A similar approach to what I suggested for C++ is needed.